pkg/asset/store/assetsfetcher.go

```diff
@@ -0,0 +1,64 @@
+package store
+
+import (
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// Engine used to retrieve and resolve a specified set of assets.
+type AssetsFetcher interface {
+	// Fetchs all the writable assets from the configured assets store.
+	Run(assets []asset.WritableAsset) error
+}
+
+type fetcher struct {
+	storeDir string
+}
```

Comment:
would it make sense to have `storeImpl` implement `AssetFetcher`? It's an existing struct that contains this data structure.

https://github.com/openshift/installer/blob/ba9211cc50cf6319609391cdaec6b98df699f05e/pkg/asset/store/store.go#L47-L51